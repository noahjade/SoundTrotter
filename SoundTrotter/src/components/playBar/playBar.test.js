import renderer from "react-test-renderer";
import { mount } from "enzyme";
import PlayBar from ".";

const { act } = renderer;

const mockPlaybackState = {
  item: {},
  is_playing: true,
};

const mockSpotify = {
  getMyCurrentPlaybackState: () => {
    return Promise.resolve(mockPlaybackState);
  },
};

test("Correctly renders playbar with pause button when a playing track is provided", async () => {
  let view;
  await act(async () => {
    view = renderer.create(<PlayBar spotify={mockSpotify} />);
  });
  expect(view).toMatchSnapshot();
});

test("Correctly renders playbar with play button when a paused track is provided", async () => {
  mockPlaybackState.is_playing = false;
  let view;
  await act(async () => {
    view = renderer.create(<PlayBar spotify={mockSpotify} />);
  });
  expect(view).toMatchSnapshot();
});

test("Correctly hides playbar when a track is not provided", async () => {
  const view = renderer.create(<PlayBar spotify={{}} />);
  expect(view).toMatchSnapshot();
});
