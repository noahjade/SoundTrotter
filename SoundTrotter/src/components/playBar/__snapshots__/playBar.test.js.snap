// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly hides playbar when a track is not provided 1`] = `
<section
  className="hidden"
>
  <div
    className="track"
  >
    <div
      className="image"
    >
      <img />
    </div>
    <div
      className="info"
    >
      <span
        className="info__title"
      />
      <span
        className="info__artists"
      />
    </div>
    <div
      className="track__controls"
    >
      <button
        className="btn"
        id="previous-btn"
        onClick={[Function]}
      >
        <svg>
          prev.svg
        </svg>
      </button>
      <button
        className="btn"
        disabled={false}
        id="play-btn"
        onClick={[Function]}
      >
        <svg>
          play.svg
        </svg>
      </button>
      <button
        className="btn"
        id="next-btn"
        onClick={[Function]}
      >
        <svg>
          next.svg
        </svg>
      </button>
    </div>
  </div>
</section>
`;

exports[`Correctly renders playbar with pause button when a playing track is provided 1`] = `
<section
  className="container"
>
  <div
    className="track"
  >
    <div
      className="image"
    >
      <img />
    </div>
    <div
      className="info"
    >
      <span
        className="info__title"
      />
      <span
        className="info__artists"
      />
    </div>
    <div
      className="track__controls"
    >
      <button
        className="btn"
        id="previous-btn"
        onClick={[Function]}
      >
        <svg>
          prev.svg
        </svg>
      </button>
      <button
        className="btn"
        disabled={false}
        id="pause-btn"
        onClick={[Function]}
      >
        <svg>
          pause.svg
        </svg>
      </button>
      <button
        className="btn"
        id="next-btn"
        onClick={[Function]}
      >
        <svg>
          next.svg
        </svg>
      </button>
    </div>
  </div>
</section>
`;

exports[`Correctly renders playbar with play button when a paused track is provided 1`] = `
<section
  className="container"
>
  <div
    className="track"
  >
    <div
      className="image"
    >
      <img />
    </div>
    <div
      className="info"
    >
      <span
        className="info__title"
      />
      <span
        className="info__artists"
      />
    </div>
    <div
      className="track__controls"
    >
      <button
        className="btn"
        id="previous-btn"
        onClick={[Function]}
      >
        <svg>
          prev.svg
        </svg>
      </button>
      <button
        className="btn"
        disabled={false}
        id="play-btn"
        onClick={[Function]}
      >
        <svg>
          play.svg
        </svg>
      </button>
      <button
        className="btn"
        id="next-btn"
        onClick={[Function]}
      >
        <svg>
          next.svg
        </svg>
      </button>
    </div>
  </div>
</section>
`;
